thorlcr/slave/thslavemain.cpp

```diff
@@ -219,16 +219,21 @@ bool ControlHandler(ahType type)
 {
     if (ahInterrupt == type)
         LOG(MCdebugProgress, thorJob, "CTRL-C detected");
-    else if (!jobListenerStopped)
-        LOG(MCdebugProgress, thorJob, "SIGTERM detected");
-    bool unregOK = false;
+    else
+    {
+        if (jobListenerStopped)
+            return false;
+        else
+            LOG(MCdebugProgress, thorJob, "SIGTERM detected");
+    }
+
     if (!jobListenerStopped)
     {
         if (masterNode)
-            unregOK = UnregisterSelf(NULL);
+            UnregisterSelf(NULL);
         abortSlave();
     }
-    return !unregOK;
+    return true;
```

Comment:
Can you explain why returning true here?
It will cause exit (_exit(0)), in the abort handler, which may be okay anyway,
but.. prior to this change it returned false if UnregisterSelf succeeded (implying a clean unregistration from master)

If it crashes then ( as the logs/stack trace indicate in the JIRA ), then that's an issue..
IOW - is this change to _exit(0) - just covering up a clean exit?

Perhaps it all makes sense, but it's not clear, if there's a reason for hard exit here, then a comment explaining why may suffice.